pkg/controller/storagecluster/cephcluster.go

```diff
@@ -475,3 +484,29 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+func (r *ReconcileStorageCluster) throttleStorageDevices(storageClassName string) (bool, error) {
```

Comment:
The previous function looks better to me - it just determines whether the storageClass should be throttled and lets the calling function handle the rest.

Reply:
this has been taken care of, afaict